Userland/Libraries/LibWeb/WebAssembly/WebAssemblyObject.cpp

```diff
@@ -89,16 +93,20 @@ JS_DEFINE_NATIVE_FUNCTION(WebAssemblyObject::instantiate)
 {
     // FIXME: This shouldn't block!
     auto buffer = vm.argument(0).to_object(global_object);
-    JS::Value exception;
-    if (vm.exception()) {
-        exception = vm.exception()->value();
-        vm.clear_exception();
-    }
     auto promise = JS::Promise::create(global_object);
-    if (!exception.is_empty()) {
-        promise->reject(exception);
+    auto take_exception_and_reject_if_needed = [&] {
+        if (vm.exception()) {
+            JS::Value exception = vm.exception()->value();
+            promise->reject(exception);
+            vm.clear_exception();
```

Comment:
Let's make sure to clear the exception first, rejecting the promise can lead to JS function calls (not right now I think, but definitely once the `unhandledrejection` event is implemented), and that would fail miserably. :)